_pytest/python_api.py

```diff
@@ -6,10 +6,271 @@
 from _pytest.compat import isclass
 from _pytest.runner import fail
 import _pytest._code
+
 # builtin pytest.approx helper
 
+class ApproxBase(object):
+    """
+    Provide shared utilities for making approximate comparisons between numbers 
+    or sequences of numbers.
+    """
+
+    def __init__(self, expected, rel=None, abs=None):
+        self.expected = expected
+        self.abs = abs
+        self.rel = rel
+
+    def __repr__(self):
+        return ', '.join(
+                repr(self._approx_scalar(x))
+                for x in self._yield_expected())
+
+    def __eq__(self, actual):
+        return all(
+                a == self._approx_scalar(x)
+                for a, x in self._yield_comparisons(actual))
+
+    __hash__ = None
+
+    def __ne__(self, actual):
+        return not (actual == self)
+
+    def _approx_scalar(self, x):
+        return ApproxScalar(x, rel=self.rel, abs=self.abs)
+
+    def _yield_expected(self, actual):
+        """
+        Yield all the expected values associated with this object.  This is 
+        used to implement the `__repr__` method.
+        """
+        raise NotImplementedError
+
+    def _yield_comparisons(self, actual):
+        """
+        Yield all the pairs of numbers to be compared.  This is used to 
+        implement the `__eq__` method.
+        """
+        raise NotImplementedError
+
+
+
+try:
+    import numpy as np
+
+    class ApproxNumpy(ApproxBase, np.ndarray):
+        """
+        Perform approximate comparisons for numpy arrays.
+
+        This class must inherit from numpy.ndarray in order to allow the approx 
```

Comment:
That's a good point. I think it is reasonable to document to users that `approx` should always be on the left side.

Btw, this wouldn't actually break anything because `numpy` arrays are not currently supported anyway. 😉